### PR TITLE
[REFACTOR] Use geopandas to calculate state/city information

### PR DESCRIPTION
**Pull request by jcalvarezj**
This PR implements reverse geocoding without using an API through the geopandas library.
It requires a geographical dataset, which has been obtained from DIVA-GIS.

### CODE REVIEW

File: etl/main.py [hunk 1]
```diff
@@ -1,214 +1,242 @@
...
+    with engine.connect() as connection:
+        print('Inserting data...')
+
+        for record in stations_dict:
+            stations_data = {
+                'id': record['id'],
+                'name': record['name'],
+                'longitude': record['longitude'],
+                'latitude': record['latitude'],
+                'town': record['city'],
```
Comment (AndresCendales) [inline etl/main.py:197]: There is a  key error when the script run. This is the complete traceback:

image

image

File: etl/main.py [hunk 2]
```diff
@@ -1,214 +1,242 @@
...
+import traceback
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import sqlalchemy as db
+import xmltodict as x2d
+from pathlib import Path
+from datetime import timedelta, datetime
+from timeit import default_timer as timer
+from dummy import get_city_and_state_for_location
```
Comment (AndresCendales) [inline etl/main.py]: Please remove this unused import

**Review by AndresCendales — CHANGES_REQUESTED** [review]

File: etl/main.py (hunk 2 above)
Comment (jcalvarezj) [reply etl/main.py]: Done

File: etl/main.py (hunk 1 above)
Comment (jcalvarezj) [reply etl/main.py:197]: Corrected.

**Review by AndresCendales — APPROVED** [review]

**Pull request merged**